用官方的create-react-app创建的项目怎么添加antd

用官方的create-react-app创建的项目怎么添加antd

现在start可以运行了，但是运行build的时候只要加上了antd的组件就build失败

```

Failed to create a production build. Reason:
Module build failed: Error: /home/lise/GitHub/react/src/App.js: File/Program node, we can't possibly find a statement parent to this
    at NodePath.getStatementParent (/home/lise/GitHub/react/node_modules/.npminstall/babel-traverse/6.13.0/babel-traverse/lib/path/family.js:43:11)
    at PathHoister.getAttachmentPath (/home/lise/GitHub/react/node_modules/.npminstall/babel-traverse/6.13.0/babel-traverse/lib/path/lib/hoister.js:118:26)
    at PathHoister.run (/home/lise/GitHub/react/node_modules/.npminstall/babel-traverse/6.13.0/babel-traverse/lib/path/lib/hoister.js:171:25)
    at NodePath.hoist (/home/lise/GitHub/react/node_modules/.npminstall/babel-traverse/6.13.0/babel-traverse/lib/path/modification.js:292:18)
    at PluginPass.JSXElement (/home/lise/GitHub/react/node_modules/babel-plugin-transform-react-constant-elements/lib/index.js:39:16)
    at newFn (/home/lise/GitHub/react/node_modules/.npminstall/babel-traverse/6.13.0/babel-traverse/lib/visitors.js:301:19)
    at NodePath._call (/home/lise/GitHub/react/node_modules/.npminstall/babel-traverse/6.13.0/babel-traverse/lib/path/context.js:76:18)
    at NodePath.call (/home/lise/GitHub/react/node_modules/.npminstall/babel-traverse/6.13.0/babel-traverse/lib/path/context.js:48:17)
    at NodePath.visit (/home/lise/GitHub/react/node_modules/.npminstall/babel-traverse/6.13.0/babel-traverse/lib/path/context.js:106:12)
    at TraversalContext.visitQueue (/home/lise/GitHub/react/node_modules/.npminstall/babel-traverse/6.13.0/babel-traverse/lib/context.js:167:16)
    at TraversalContext.visitSingle (/home/lise/GitHub/react/node_modules/.npminstall/babel-traverse/6.13.0/babel-traverse/lib/context.js:118:19)
    at TraversalContext.visit (/home/lise/GitHub/react/node_modules/.npminstall/babel-traverse/6.13.0/babel-traverse/lib/context.js:211:19)
    at Function.traverse.node (/home/lise/GitHub/react/node_modules/.npminstall/babel-traverse/6.13.0/babel-traverse/lib/index.js:161:17)
    at NodePath.visit (/home/lise/GitHub/react/node_modules/.npminstall/babel-traverse/6.13.0/babel-traverse/lib/path/context.js:116:19)
    at TraversalContext.visitQueue (/home/lise/GitHub/react/node_modules/.npminstall/babel-traverse/6.13.0/babel-traverse/lib/context.js:167:16)
    at TraversalContext.visitMultiple (/home/lise/GitHub/react/node_modules/.npminstall/babel-traverse/6.13.0/babel-traverse/lib/context.js:113:17)
    at TraversalContext.visit (/home/lise/GitHub/react/node_modules/.npminstall/babel-traverse/6.13.0/babel-traverse/lib/context.js:209:19)

npm ERR! Linux 4.4.0-31-generic
npm ERR! argv "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/bin/npm-cli.js" "run-script" "build"
npm ERR! node v6.3.1
npm ERR! npm  v3.10.3
npm ERR! code ELIFECYCLE
npm ERR! react@0.0.1 build: `node ./scripts/build.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the react@0.0.1 build script 'node ./scripts/build.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the react package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ./scripts/build.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs react
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls react
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/lise/GitHub/react/npm-debug.log
```

--- a/docs/react/customize-theme.en-US.md
+++ b/docs/react/customize-theme.en-US.md
@@ -1,5 +1,5 @@
 ---
-order: 4
+order: 5
 title: Customize Theme
 ---
 

--- a/docs/react/customize-theme.zh-CN.md
+++ b/docs/react/customize-theme.zh-CN.md
@@ -1,5 +1,5 @@
 ---
-order: 4
+order: 5
 title: 定制主题
 ---
 

--- a/docs/react/i18n-solution.en-US.md
+++ b/docs/react/i18n-solution.en-US.md
@@ -1,5 +1,5 @@
 ---
-order: 5
+order: 6
 title: i18n Solution
 link: //github.com/ant-design/intl-example
 ---

--- a/docs/react/i18n-solution.zh-CN.md
+++ b/docs/react/i18n-solution.zh-CN.md
@@ -1,5 +1,5 @@
 ---
-order: 5
+order: 6
 title: i18n 方案
 link: //github.com/ant-design/intl-example
 ---

--- a/docs/react/practical-projects.en-US.md
+++ b/docs/react/practical-projects.en-US.md
@@ -1,6 +1,6 @@
 ---
 order: 3
-title: Practical Projects
+title: Real world Example with dva
 ---
 
 [dva](https://github.com/dvajs/dva) is a React and redux based, lightweight and elm-style framework, which supports side effects, hot module replacement, dynamic on demand, react-native, SSR. And it has been widely used in production environment.

--- a/docs/react/use-with-create-react-app.en-US.md
+++ b/docs/react/use-with-create-react-app.en-US.md
@@ -171,13 +171,13 @@ loaders: [
     query: {
       plugins: [
 -       ['import', [{ libraryName: "antd", style: 'css' }]],
-+       ['import', [{ libraryName: "antd", style: true }]],  // 加载 less 文件
++       ['import', [{ libraryName: "antd", style: true }]],  // import less
       ],
    },
 
 ...
 
-+ // 解析 less 文件，并加入变量覆盖配置
++ // Parse less files and modify variables
 + {
 +   test: /\.less$/,
 +   loader: 'style!css!postcss!less?{modifyVars:{"@primary-color":"#1DA57A"}}'

--- a/docs/react/use-with-create-react-app.en-US.md
+++ b/docs/react/use-with-create-react-app.en-US.md
@@ -31,7 +31,7 @@ $ cd antd-demo
 $ npm start
 ```
 
-Open browser at http://localhost:3000/ , everything is going fine when you see `Welcome to React`.
+Open browser at http://localhost:3000/, it renders a header saying "Welcome to React" on the page.
 
 ## Import antd
 
@@ -185,7 +185,7 @@ loaders: [
 ]
 ```
 
-We use `modifyVars` option of [less-loader](https://github.com/webpack/less-loader#less-options). If you see a green button after reboot `npm start`, then everything is working now.
+We use `modifyVars` option of [less-loader](https://github.com/webpack/less-loader#less-options) here, you can see a green button rendered on the page after reboot start server.
 
 ---
 

--- a/docs/react/use-with-create-react-app.en-US.md
+++ b/docs/react/use-with-create-react-app.en-US.md
@@ -3,38 +3,39 @@ order: 4
 title: Use with create-react-app
 ---
 
-[create-react-app](https://github.com/facebookincubator/create-react-app) 是业界最优秀的 React 应用开发工具之一，本文会尝试在 create-react-app 创建的工程中使用 antd 组件，并自定义 webpack 的配置以满足各类工程化需求。
+[create-react-app](https://github.com/facebookincubator/create-react-app) is one of best React application development tool, we are going to use `antd` within it and modify the webpack config for some customized needs.
 
 ---
 
-## 安装和初始化
+## Install and Initialization
 
-我们需要在命令行中安装 create-react-app 工具。
+We need to install `create-react-app` first.
 
 ```bash
 $ npm install -g create-react-app
 ```
 
-然后新建一个项目。
+Create a new project named `antd-demo`.
 
 ```bash
 $ USE_YARN=no create-react-app antd-demo
 ```
 
-工具会自动初始化一个脚手架并安装 React 项目的各种必要依赖，如果在过程中出现网络问题，请尝试配置代理或使用其他 npm registry。
+The tool will create and initialize environment and dependencies automaticly,
+please try config your proxy setting or use other npm registry if any network errors happen during it.
 
-然后我们进入项目并启动。
+Then we go inside `antd-demo` and start it.
 
 ```bash
 $ cd antd-demo
 $ npm start
 ```
 
-此时浏览器会访问 http://localhost:3000/ ，看到 `Welcome to React` 的界面就算成功了。
+Open browser at http://localhost:3000/ , everything is going fine when you see `Welcome to React`.
 
-## 引入 antd
+## Import antd
 
-这是 create-react-app 生成的默认目录结构。
+It is the default directory structure below.
 
 ```
 ├── README.md
@@ -52,13 +53,13 @@ $ npm start
 └── yarn.lock
 ```
 
-现在从 npm 安装并引入 antd。
+Now we install `antd` from npm.
 
 ```bash
 $ npm install antd --save
 ```
 
-修改 `src/App.js`，引入 antd 的按钮组件。
+Modify `src/App.js`, import Button component from `antd`.
 
 ```jsx
 import React, { Component } from 'react';
@@ -78,7 +79,7 @@ class App extends Component {
 export default App;
 ```
 
-修改 `src/App.css`，在文件顶部引入 `antd/dist/antd.css`。
+Add `antd/dist/antd.css` at the top of `src/App.css`.
 
 ```css
 @import '~antd/dist/antd.css';
@@ -90,25 +91,27 @@ export default App;
 ...
 ```
 
-好了，现在你应该能看到页面上已经有了 antd 的蓝色按钮组件，接下来就可以继续选用其他组件开发应用了。其他开发流程你可以参考 create-react-app 的[官方文档](https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/template/README.md)。
+Ok, you now see a blue primary button displaying in page now, next you can choose any components of `antd` to develop your application. Visit other workflow of `create-react-app` at its [User Guide ](https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/template/README.md).
 
-## 高级配置
 
-我们现在已经把组件成功运行起来了，但是在实际开发过程中还有很多问题，例如上面的例子实际上加载了全部的 antd 组件的代码（对前端性能是个隐患）。
+## Advanced Guides
 
-> 你会在控制台看到如下警告。
+We are successd to run antd components now, but in the real world, there are still lots of problems about antd-demo.
+For instance, we actully import all components in the project which will be a serious network perfermance issue.
+
+> You will see a warning in your browser console.
 
 ![](https://zos.alipayobjects.com/rmsportal/dBLScZPjiUwunfyQVISX.png)
 
-我们需要对 create-react-app 的默认配置进行自定义。可以使用 `eject` 命令将所有内建的配置暴露出来。
+So it is necessary to customize the default webpack config. We can achieve that by using `eject` script command.
 
 ```bash
 $ npm run eject
 ```
 
-### 按需加载
+### Import on demand
 
-现在可以安装 [babel-plugin-import](https://github.com/ant-design/babel-plugin-import) 并修改 `src/webconfig.dev.js` 文件。
+After eject all config files to antd-demo, we allowed to install [babel-plugin-import](https://github.com/ant-design/babel-plugin-import) and modify `src/webconfig.dev.js` now.
 
 ```bash
 $ npm install babel-plugin-import --save-dev
@@ -132,13 +135,13 @@ $ npm install babel-plugin-import --save-dev
 },
 ```
 
-然后移除前面在 `src/App.css` 里全量添加的 `@import '~antd/dist/antd.css';` 样式代码，现在 babel-plugin-import 会按需加载样式。
+Remove the `@import '~antd/dist/antd.css';` statement added before because `babel-plugin-import` will import styles.
 
-最后重启 `npm start` 访问页面，此时上面的警告信息应该没了，antd 组件的 js 和 css 代码都会按需加载。
+Then reboot `npm start` and visit demo page, you should find that the above warning message would be gone which prove the `import on demand` config is effective now.
 
-### 自定义主题
+### Customize Theme
 
-按照 [配置主题](/docs/react/customize-theme) 的要求，自定义主题需要用到 less 变量覆盖功能，因此首先我们需要引入 [less-loader](https://github.com/webpack/less-loader) 来加载 less 样式，同时修改 `src/webconfig.dev.js` 文件。
+According to [Customize Theme documentation](/docs/react/customize-theme), we need `less` variables modify ability of [less-loader](https://github.com/webpack/less-loader), so we add it.
 
 ```bash
 $ npm install less less-loader --save-dev
@@ -182,13 +185,10 @@ loaders: [
 ]
 ```
 
-这里利用了 [less-loader](https://github.com/webpack/less-loader#less-options) 的 `modifyVars` 来进行主题配置，
-变量和其他配置方式可以参考 [配置主题](/docs/react/customize-theme) 文档。
-
-修改后重启 `npm start`，如果看到一个绿色的按钮就说明配置成功了。
+We use `modifyVars` option of [less-loader](https://github.com/webpack/less-loader#less-options). If you see a green button after reboot `npm start`, then everything is working now.
 
 ---
 
-以上是在 create-react-app 中使用 antd 的相关实践，你也可以借鉴此文的做法在自己的 webpack 工作流中使用 antd，更多 webpack 配置可参考 [atool-build](https://github.com/ant-tool/atool-build/blob/master/src/getWebpackCommonConfig.js)。
+Finally, we use antd with create-react-app successfully, you can learn these practice for your own webpack workflow too, and find more webpack config in the [atool-build](https://github.com/ant-tool/atool-build/blob/master/src/getWebpackCommonConfig.js).
 
-本例源码可见：https://github.com/ant-design/create-react-app-antd
+Source code about this article：https://github.com/ant-design/create-react-app-antd

--- a/docs/react/use-with-create-react-app.en-US.md
+++ b/docs/react/use-with-create-react-app.en-US.md
@@ -1,0 +1,194 @@
+---
+order: 4
+title: Use with create-react-app
+---
+
+[create-react-app](https://github.com/facebookincubator/create-react-app) 是业界最优秀的 React 应用开发工具之一，本文会尝试在 create-react-app 创建的工程中使用 antd 组件，并自定义 webpack 的配置以满足各类工程化需求。
+
+---
+
+## 安装和初始化
+
+我们需要在命令行中安装 create-react-app 工具。
+
+```bash
+$ npm install -g create-react-app
+```
+
+然后新建一个项目。
+
+```bash
+$ USE_YARN=no create-react-app antd-demo
+```
+
+工具会自动初始化一个脚手架并安装 React 项目的各种必要依赖，如果在过程中出现网络问题，请尝试配置代理或使用其他 npm registry。
+
+然后我们进入项目并启动。
+
+```bash
+$ cd antd-demo
+$ npm start
+```
+
+此时浏览器会访问 http://localhost:3000/ ，看到 `Welcome to React` 的界面就算成功了。
+
+## 引入 antd
+
+这是 create-react-app 生成的默认目录结构。
+
+```
+├── README.md
+├── package.json
+├── public
+│   ├── favicon.ico
+│   └── index.html
+├── src
+│   ├── App.css
+│   ├── App.js
+│   ├── App.test.js
+│   ├── index.css
+│   ├── index.js
+│   └── logo.svg
+└── yarn.lock
+```
+
+现在从 npm 安装并引入 antd。
+
+```bash
+$ npm install antd --save
+```
+
+修改 `src/App.js`，引入 antd 的按钮组件。
+
+```jsx
+import React, { Component } from 'react';
+import { Button } from 'antd';
+import './App.css';
+
+class App extends Component {
+  render() {
+    return (
+      <div className="App">
+        <Button type="primary">Button</Button>
+      </div>
+    );
+  }
+}
+
+export default App;
+```
+
+修改 `src/App.css`，在文件顶部引入 `antd/dist/antd.css`。
+
+```css
+@import '~antd/dist/antd.css';
+
+.App {
+  text-align: center;
+}
+
+...
+```
+
+好了，现在你应该能看到页面上已经有了 antd 的蓝色按钮组件，接下来就可以继续选用其他组件开发应用了。其他开发流程你可以参考 create-react-app 的[官方文档](https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/template/README.md)。
+
+## 高级配置
+
+我们现在已经把组件成功运行起来了，但是在实际开发过程中还有很多问题，例如上面的例子实际上加载了全部的 antd 组件的代码（对前端性能是个隐患）。
+
+> 你会在控制台看到如下警告。
+
+![](https://zos.alipayobjects.com/rmsportal/dBLScZPjiUwunfyQVISX.png)
+
+我们需要对 create-react-app 的默认配置进行自定义。可以使用 `eject` 命令将所有内建的配置暴露出来。
+
+```bash
+$ npm run eject
+```
+
+### 按需加载
+
+现在可以安装 [babel-plugin-import](https://github.com/ant-design/babel-plugin-import) 并修改 `src/webconfig.dev.js` 文件。
+
+```bash
+$ npm install babel-plugin-import --save-dev
+```
+
+```diff
+// Process JS with Babel.
+{
+  test: /\.(js|jsx)$/,
+  include: paths.appSrc,
+  loader: 'babel',
+  query: {
++   plugins: [
++     ['import', [{ libraryName: "antd", style: 'css' }]],
++   ],
+    // This is a feature of `babel-loader` for webpack (not Babel itself).
+    // It enables caching results in ./node_modules/.cache/babel-loader/
+    // directory for faster rebuilds.
+    cacheDirectory: true
+  }
+},
+```
+
+然后移除前面在 `src/App.css` 里全量添加的 `@import '~antd/dist/antd.css';` 样式代码，现在 babel-plugin-import 会按需加载样式。
+
+最后重启 `npm start` 访问页面，此时上面的警告信息应该没了，antd 组件的 js 和 css 代码都会按需加载。
+
+### 自定义主题
+
+按照 [配置主题](/docs/react/customize-theme) 的要求，自定义主题需要用到 less 变量覆盖功能，因此首先我们需要引入 [less-loader](https://github.com/webpack/less-loader) 来加载 less 样式，同时修改 `src/webconfig.dev.js` 文件。
+
+```bash
+$ npm install less less-loader --save-dev
+```
+
+```diff
+loaders: [
+  {
+    exclude: [
+      /\.html$/,
+      /\.(js|jsx)$/,
++     /\.less$/,
+      /\.css$/,
+      /\.json$/,
+      /\.svg$/
+    ],
+    loader: 'url',
+  },
+
+...
+
+  // Process JS with Babel.
+  {
+    test: /\.(js|jsx)$/,
+    include: paths.appSrc,
+    loader: 'babel',
+    query: {
+      plugins: [
+-       ['import', [{ libraryName: "antd", style: 'css' }]],
++       ['import', [{ libraryName: "antd", style: true }]],  // 加载 less 文件
+      ],
+   },
+
+...
+
++ // 解析 less 文件，并加入变量覆盖配置
++ {
++   test: /\.less$/,
++   loader: 'style!css!postcss!less?{modifyVars:{"@primary-color":"#1DA57A"}}'
++ },
+]
+```
+
+这里利用了 [less-loader](https://github.com/webpack/less-loader#less-options) 的 `modifyVars` 来进行主题配置，
+变量和其他配置方式可以参考 [配置主题](/docs/react/customize-theme) 文档。
+
+修改后重启 `npm start`，如果看到一个绿色的按钮就说明配置成功了。
+
+---
+
+以上是在 create-react-app 中使用 antd 的相关实践，你也可以借鉴此文的做法在自己的 webpack 工作流中使用 antd，更多 webpack 配置可参考 [atool-build](https://github.com/ant-tool/atool-build/blob/master/src/getWebpackCommonConfig.js)。
+
+本例源码可见：https://github.com/ant-design/create-react-app-antd

--- a/docs/react/use-with-create-react-app.zh-CN.md
+++ b/docs/react/use-with-create-react-app.zh-CN.md
@@ -1,0 +1,194 @@
+---
+order: 4
+title: 在 create-react-app 中使用
+---
+
+[create-react-app](https://github.com/facebookincubator/create-react-app) 是业界最优秀的 React 应用开发工具之一，本文会尝试在 create-react-app 创建的工程中使用 antd 组件，并自定义 webpack 的配置以满足各类工程化需求。
+
+---
+
+## 安装和初始化
+
+我们需要在命令行中安装 create-react-app 工具。
+
+```bash
+$ npm install -g create-react-app
+```
+
+然后新建一个项目。
+
+```bash
+$ USE_YARN=no create-react-app antd-demo
+```
+
+工具会自动初始化一个脚手架并安装 React 项目的各种必要依赖，如果在过程中出现网络问题，请尝试配置代理或使用其他 npm registry。
+
+然后我们进入项目并启动。
+
+```bash
+$ cd antd-demo
+$ npm start
+```
+
+此时浏览器会访问 http://localhost:3000/ ，看到 `Welcome to React` 的界面就算成功了。
+
+## 引入 antd
+
+这是 create-react-app 生成的默认目录结构。
+
+```
+├── README.md
+├── package.json
+├── public
+│   ├── favicon.ico
+│   └── index.html
+├── src
+│   ├── App.css
+│   ├── App.js
+│   ├── App.test.js
+│   ├── index.css
+│   ├── index.js
+│   └── logo.svg
+└── yarn.lock
+```
+
+现在从 npm 安装并引入 antd。
+
+```bash
+$ npm install antd --save
+```
+
+修改 `src/App.js`，引入 antd 的按钮组件。
+
+```jsx
+import React, { Component } from 'react';
+import { Button } from 'antd';
+import './App.css';
+
+class App extends Component {
+  render() {
+    return (
+      <div className="App">
+        <Button type="primary">Button</Button>
+      </div>
+    );
+  }
+}
+
+export default App;
+```
+
+修改 `src/App.css`，在文件顶部引入 `antd/dist/antd.css`。
+
+```css
+@import '~antd/dist/antd.css';
+
+.App {
+  text-align: center;
+}
+
+...
+```
+
+好了，现在你应该能看到页面上已经有了 antd 的蓝色按钮组件，接下来就可以继续选用其他组件开发应用了。其他开发流程你可以参考 create-react-app 的[官方文档](https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/template/README.md)。
+
+## 高级配置
+
+我们现在已经把组件成功运行起来了，但是在实际开发过程中还有很多问题，例如上面的例子实际上加载了全部的 antd 组件的代码（对前端性能是个隐患）。
+
+> 你会在控制台看到如下警告。
+
+![](https://zos.alipayobjects.com/rmsportal/dBLScZPjiUwunfyQVISX.png)
+
+我们需要对 create-react-app 的默认配置进行自定义。可以使用 `eject` 命令将所有内建的配置暴露出来。
+
+```bash
+$ npm run eject
+```
+
+### 按需加载
+
+现在可以安装 [babel-plugin-import](https://github.com/ant-design/babel-plugin-import) 并修改 `src/webconfig.dev.js` 文件。
+
+```bash
+$ npm install babel-plugin-import --save-dev
+```
+
+```diff
+// Process JS with Babel.
+{
+  test: /\.(js|jsx)$/,
+  include: paths.appSrc,
+  loader: 'babel',
+  query: {
++   plugins: [
++     ['import', [{ libraryName: "antd", style: 'css' }]],
++   ],
+    // This is a feature of `babel-loader` for webpack (not Babel itself).
+    // It enables caching results in ./node_modules/.cache/babel-loader/
+    // directory for faster rebuilds.
+    cacheDirectory: true
+  }
+},
+```
+
+然后移除前面在 `src/App.css` 里全量添加的 `@import '~antd/dist/antd.css';` 样式代码，现在 babel-plugin-import 会按需加载样式。
+
+最后重启 `npm start` 访问页面，此时上面的警告信息应该没了，antd 组件的 js 和 css 代码都会按需加载。
+
+### 自定义主题
+
+按照 [配置主题](/docs/react/customize-theme) 的要求，自定义主题需要用到 less 变量覆盖功能，因此首先我们需要引入 [less-loader](https://github.com/webpack/less-loader) 来加载 less 样式，同时修改 `src/webconfig.dev.js` 文件。
+
+```bash
+$ npm install less less-loader --save-dev
+```
+
+```diff
+loaders: [
+  {
+    exclude: [
+      /\.html$/,
+      /\.(js|jsx)$/,
++     /\.less$/,
+      /\.css$/,
+      /\.json$/,
+      /\.svg$/
+    ],
+    loader: 'url',
+  },
+
+...
+
+  // Process JS with Babel.
+  {
+    test: /\.(js|jsx)$/,
+    include: paths.appSrc,
+    loader: 'babel',
+    query: {
+      plugins: [
+-       ['import', [{ libraryName: "antd", style: 'css' }]],
++       ['import', [{ libraryName: "antd", style: true }]],  // 加载 less 文件
+      ],
+   },
+
+...
+
++ // 解析 less 文件，并加入变量覆盖配置
++ {
++   test: /\.less$/,
++   loader: 'style!css!postcss!less?{modifyVars:{"@primary-color":"#1DA57A"}}'
++ },
+]
+```
+
+这里利用了 [less-loader](https://github.com/webpack/less-loader#less-options) 的 `modifyVars` 来进行主题配置，
+变量和其他配置方式可以参考 [配置主题](/docs/react/customize-theme) 文档。
+
+修改后重启 `npm start`，如果看到一个绿色的按钮就说明配置成功了。
+
+---
+
+以上是在 create-react-app 中使用 antd 的相关实践，你也可以借鉴此文的做法在自己的 webpack 工作流中使用 antd，更多 webpack 配置可参考 [atool-build](https://github.com/ant-tool/atool-build/blob/master/src/getWebpackCommonConfig.js)。
+
+本例源码可见：https://github.com/ant-design/create-react-app-antd

